tools/upgrade_versions.py

```diff
@@ -243,6 +254,7 @@ def run(CONF):
         pkg_version.update(committer_name, committer_email)
         pkg.unlock()
 
-    release_date = datetime.today().date().isoformat()
-    push_new_versions(versions_repo, release_date, push_repo_url,
-                      push_repo_branch, committer_name, committer_email)
+    if push_repo:
+        release_date = datetime.today().date().isoformat()
+        push_new_versions(versions_repo, release_date, push_repo_url,
+                          push_repo_branch, committer_name, committer_email)
```

Comment:
We still want a local commit, so either we put this if inside the `push_new_versions` function or split into two, e.g. `create_commit` and `push_new_versions`. I think the latter will look better.
This local commit will then have to be passed to the `build_host_os` job in the nightly job.